packages/library/src/commonMain/kotlin/io/realm/internal/SuspendableNotifier.kt

```diff
@@ -245,6 +270,42 @@ internal class SuspendableNotifier(private val owner: Realm, private val dispatc
         return NotificationToken(callback, token)
     }
 
+    /**
+     * Listen to changes to a RealmList through a change listener. The callback will happen
+     * on the configured [SuspendableNotifier.dispatcher] thread.
+     *
+     * FIXME Callers of this method must make sure it is called on the correct [SuspendableNotifier.dispatcher].
+     */
+    internal fun <T> registerListChangedListener(
```

Comment:
Again exactly the same code. I don't see anything specific for lists. Can't we generic-alize it.